### DISCUSSION TITLE
After doing  page.insert_htmlbox to insert some text , how to extract this same text with get_text where the place it has been inserted is respected.

### DISCUSSION THREAD

**Discussion by 808Code** [pymupdf/PyMuPDF#3283] [Looking for help] [answered]
I did 
`
    page.insert_htmlbox(widget.rect, f"{index}", overlay = True) 
`

Where i wanted to place a number where pdf has input fields.

Visual of what i did: 
![download (1)](https://github.com/pymupdf/PyMuPDF/assets/75775205/a5cb486b-f308-4f54-8fa2-8f7b873e7c83)

But when i do  `page.get_text` all the numbers are ignored .

What i wanted was/ Desired Output :

 `.......Purpose of Loan? 0 Purchase 1 Construction 2 Refinance 3 Other  ......`

What it gives me:

` ....Purpose of Loan?  Purchase  Construction  Refinance  Other ....` 

and at the end of the text ,  all number from` 0 to ..... total input field ` is given.

How to get my desired output. 

Thankyou for any help.

**Comment by JorjMcKie** [comment]
All text extraction and text search variants support the `clip` parameter: a rect_like.
So if you used `page.get_text(clip=rect)` where rect is the rectangle used with `insert_htmlbox` you should receive that inserted text.

**Reply by 808Code** [reply]
Yes , when i do `page.get_text(clip= widget.rect)` .I get the text associated with the widget , but i want the text to be extracted the same way `get_text` with `clip = None`. works i.e.. extracted text is in the  same order as it appears in the pdf. where  widget text  is beside whatever the normal pdf text is along.

**Comment by JorjMcKie** [comment, marked as answer]
> Yes , when i do `page.get_text(clip= widget.rect)` .I get the text associated with the widget , but i want the text to be extracted the same way `get_text` with `clip = None`. works i.e.. extracted text is in the same order as it appears in the pdf. where widget text is beside whatever the normal pdf text is along.

This **_cannot_** work!
All new stuff in a PDF can only be appended to old content - not inserted in the middle of things by some miracle. This a PDF peculiarity - not a PyMuPDF restriction.
The only way you have is sorting extracted text in a suitable way.
There is a `sort` parameter in get_text()  which behaves slightly differently depending on the output option.
In your case however - as you are probably inserting "words" in some way - you probably have to `get_text("words", sort=True)` and then process the single words, re-synthesizing lines etc.

**Reply by 808Code** [reply]
Oh oh .Thankyou.
